sys/include/net/sock/dodtls.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_dodtls     DNS over DTLS sock API
+ * @ingroup     net_sock
+ *
+ * @brief       Sock DNS over DTLS client
+ *
+ * @see         [RFC 8094](https://datatracker.ietf.org/doc/html/rfc8094)
+ *
+ * @experimental    This implementation is in an experimental state.
+ *                  RFC 8094 requires DNS over TLS (DoT) as a fall-back for the
+ *                  [PMTU issues](https://datatracker.ietf.org/doc/html/rfc8094#section-5)).
+ *                  This fallback is not in place in this implementation.
+ *                  Consequently, [EDNS(0)](https://datatracker.ietf.org/doc/html/rfc6891)
+ *                  to negotiate maximum response size is also not in place.
+ *
+ * @{
+ *
+ * @file
+ * @brief   DNS over DTLS sock definitions
+ *
+ * @author  Martine S. Lenders <m.lenders@fu-berlin.de>
+ */
+
+#ifndef NET_SOCK_DODTLS_H
+#define NET_SOCK_DODTLS_H
+
+#include "net/sock/dtls.h"
+#include "net/sock/udp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name DNS over DTLS defines
+ * @{
+ */
+#define SOCK_DODTLS_PORT                    (853)   /**< Default DNS over DTLS server port */
+
+/**
+ * @defgroup    net_sock_dodtls_conf DNS over DTLS compile-time configuration
+ * @ingroup     config
+ * @{
+ */
+#ifndef CONFIG_SOCK_DODTLS_RETRIES
+#define CONFIG_SOCK_DODTLS_RETRIES          (2)     /**< Number of DNS over DTLS query retries */
+#endif
+
+/**
+ * @brief   Timeout for DNS over DTLS queries in milliseconds
+ */
+#ifndef CONFIG_SOCK_DODTLS_TIMEOUT_MS
+#define CONFIG_SOCK_DODTLS_TIMEOUT_MS       (1000U)
+#endif
+/** @} */
+
+/**
+ * @brief   Maximum name length for a DNS over DTLS query
+ */
+#define SOCK_DODTLS_MAX_NAME_LEN            (CONFIG_DNS_MSG_LEN - sizeof(dns_hdr_t) - 4)
```

Comment:
Why the 4 bytes?

Reply:
As with normal DTLS: The DNS query type (2 bytes) and class (2 bytes) in the query.